src/libs/actions/Report.js

```diff
@@ -379,11 +379,26 @@ function setLocalLastRead(reportID, sequenceNumber) {
 
     // Update the report optimistically
     Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-        unreadActionCount: 0,
+        unreadActionCount: Math.max(reportMaxSequenceNumbers[reportID] - sequenceNumber, 0),
         lastVisitedTimestamp: Date.now(),
     });
 }
 
+/**
+ * Set the New marker for a given report to the oldest unread message
```

Comment:
This description and the name of the method seem to be saying two very different things. Can you please update the description to be more clear? Practically speaking, I'm not sure what "the oldest unread message" has to do with "marker initial position" and they seem like very separate things.

Reply:
Hmmm. The description is accurate, though. That is what the method is doing:
- A: We get the total number of sequences from reportMaxSequenceNumbers[reportID] 
- B: We get how many of those are unread from allReports[reportID].unreadActionCount
- We determine where the last read one is by deducting B from A
- We set the `New` marker above the **next** sequence, which would be the first (oldest) unread one.

Does that make sense?

Reply:
It only sort-of makes sense... but I think we need to first clear up some terms. You've referred to "last unread", "first unread" and now also "oldest unread". Can we standardize on those? Is oldest unread different than the first unread?

Reply:
Gone!